### ISSUE TITLE
[META] Xcode 15.3 Support for .NET 8 and MAUI

### ISSUE THREAD

**Issue by jfversluis** [dotnet/maui#21057]
_Below text is copied from the .NET for iOS repository. TL;DR Xcode 15.3 is not supported at this time._

--- 

Hello!

Xcode 15.3 was released on March 5, 2024. We are in the process of updating our bindings and our infra in order to support this release.

As of the time of this post, Xcode 15.3 is not recommended to be used with our .NET MAUI or Xamarin SDKs so our advise is not to install it just yet and keep using Xcode 15.2 for the time being.

[If you need to download Xcode 15.2 you can get it from the Apple Developer portal.](https://developer.apple.com/download/all/?q=xcode%2015)

Known issues
iOS 17.4 Simulator crashes

https://github.com/dotnet/runtime/issues/98941
Workaround: Use Xcode 15.2. If you really need to use Xcode 15.3 then either use iOS Simulator 17.2 or the workaround provided in https://github.com/dotnet/runtime/issues/98941.

We'll keep this post updated as we learn more.

Thank you!

-The macios team (and .NET MAUI team 😄 )

**Comment by MitchBomcanhao** [comment]
this is highly necessary - a number of other issues make using 15.3 necessary (can't even debug the iOS app without it), but now it seems that a bunch of bits are entirely broken when using 15.3, in my case mostly around globalization.

**Comment by stephen-hawley** [comment]
When this is fixed, please update [.github/DEVELOPMENT.md](.github/DEVELOPMENT.md) so the dev guide doesn't contraindicate Xcode 15.3.

**Comment by Phenek** [comment]
It should be great to have this informations on the .Net MAUI Output of visual studio code!
We already have Android sdk version displayed

Output:
```shell
Android components:
	- Java SDK (version: '11.0.22'): installed.
 
	Android SDK recommended required components:
	- platforms/android-34 (version: '3'): installed.
	- build-tools/34.0.0 (version: '34.0.0'): installed.
	- platform-tools (version: '35.0.1'): installed.
	- cmdline-tools/11.0 (version: '11.0'): installed.
 
	Android SDK recommended optional components:
	- emulator (version: '34.1.19'): installed.
	- system-images/android-34/google_apis/x86_64 (version: '8'): installed.
	- system-images/android-33/google_apis_playstore/x86_64 (version: '7'): installed.

```

We could add informations about compatibilities about Xcode / Android sdk?
Orange or Red flag on this version, should be inform us.
A link with the current issue to follow on github would be the great!

**Comment by AndreKraemer** [comment]
@AOgai, I'm curious, have you attempted to run a release build on your local emulator to see if it also crashes? It's worth investigating because crashes upon app launch often occur in release builds (but not in debug builds) when the interpreter isn't utilized. For further insights, you might want to check out these issues: 
- [Issue 11276](https://github.com/dotnet/maui/issues/11276).
- [Issue 18870](https://github.com/dotnet/maui/issues/18870)

**Comment by jdarwood007** [comment]
> Edit: I think <_MonoRuntimeComponentLinking Include="static" RuntimeIdentifier="ios-arm64" /> is maybe not needed.

With that line included I receive the following error when trying to publish:
```
Value of _MonoRuntimeComponentLinking for 'ios-arm64' must be 'dynamic' or 'static' it is 'static;static'. Malformed runtime pack?
```
I had to remove that line to get it to publish.

**Comment by pierre01** [comment]
Please Fix this ASAP!!

**Comment by JoacimWall** [comment]
Hi
Use this one if you want to have multiple version of Xcode or just 2 minutes install instead of 30 minutes. 

https://www.xcodes.app
 
//Joacim

**Comment by MitchBomcanhao** [comment]
15.4 beta 1 has now been released for a few days. by the time whatever fix is done to get 15.3 done, 15.4 or 15.5 or who knows which version will already be available and probably also not working.

**Comment by stemig** [comment]
Unacceptable response times...

**Comment by malsabi** [comment]
Is there any progress related to this issue ? Our production app does not work in 15.3 and users are complaining it's crashing.

**Comment by mattlovell** [comment]
Using VS Version 17.10.0 Preview 5.0 / xcode 15.2
upgraded project to .net 8

I've been battling getting a release onto testflight for about a week now. Debugging with ios17.2 simulators in debug and release mode = no errors. Android 33 - no errors

Running on actual device using ios17.4  in release (hot restart) crashes every time as do the 17.4 simulators...

As I dont have any apple devices running 17.2 this is a bit awkward.


So any idea when the 17.4 simulators might be ready?

**Comment by mattlovell** [comment]
I managed to get an App onto testflight after removing all the CSS from the project, which turned out to be the cause of weird xaml errors that would crash the app on device or release testing...... not sure why CSS is no longer valid as worked fine for 6 months...  

Oh well only took 1.5 weeks to find the issue and missed a client deadline

**Comment by imuller** [comment]
@jfversluis Can you give an update when this is available?

**Comment by jfversluis** [comment]
@imuller subscribe to this issue and when it's closed there probably will be a comment with an update and then it will probably be available as well of have information when it will be!

**Comment by Richard-Starrtec** [comment]
Do we have an ETA on this?

**Comment by zacharykeeping** [comment]
Can confirm .NET 8.0.5 fixes this for us.

**Comment by matouskozak** [comment]
> Can confirm .NET 8.0.5 fixes this for us.

Thank you for trying the new servicing release. The .NET 8.0.5 contains the fix for https://github.com/dotnet/runtime/issues/98941, thanks for verifying that it works. This should unblock the Xcode 15.3 support. FYI: @jfversluis

**Comment by malsabi** [comment]
I updated to .NET 8.0.5 and It's still crashing on 17.4 Simulator but on 15.5 it's working completely fine.

**Comment by MAUIoxo** [comment]
So, is it already possible to do the update to 15.3 using latest nightly builds, or still wait?

**Comment by HansenPeter** [comment]
With Xcode 15.4 now available, can you provide us with an update on this please?

**Comment by somoreingold** [comment]
> Does this affect running apps on android emulators on Mac?

No, this issue is about iOS. I've got nothing but sympathy for the small team at MS working on this, but would love to know some order-of-magnitude timeline as we devs are bound to such timelines to our own bosses. Is it weeks? months? days?

**Comment by kpespisa** [comment]
I still experience a crash in my MAUI app using Xcode 15.3 or 15.4. 

` *** error for object 0x127d3fe64: pointer being freed was not allocated
<my_app_name>(62073,0x307ed4000) malloc: *** set a breakpoint in malloc_error_break to debug`

Moving back to using Xcode 15.2 avoids the problem. 

I haven't yet narrowed down what triggers the crash, but it is not occurring at a certain point in my MAUI code - it occurs at various times during startup.

I will try to find the source of the issue, probably by removing startup related code, xaml, etc. until it doesn't crash, and will open a new bug if I can get to that point. However, adding this in case anyone else has hit the same mysterious problem.

**Comment by matouskozak** [comment]
> I still experience a crash in my MAUI app using Xcode 15.3 or 15.4.
> 
> ` *** error for object 0x127d3fe64: pointer being freed was not allocated <my_app_name>(62073,0x307ed4000) malloc: *** set a breakpoint in malloc_error_break to debug`
> 
> Moving back to using Xcode 15.2 avoids the problem.
> 
> I haven't yet narrowed down what triggers the crash, but it is not occurring at a certain point in my MAUI code - it occurs at various times during startup.
> 
> I will try to find the source of the issue, probably by removing startup related code, xaml, etc. until it doesn't crash, and will open a new bug if I can get to that point. However, adding this in case anyone else has hit the same mysterious problem.

This looks like a similar error to this one, however we would need to see the stack trace to confirm. You can check https://github.com/dotnet/runtime/issues/98941 to see if the stack trace looks similar. If that is the case, I would recommend verifying that your MAUI app is using the 8.0.5 .NET runtime as it might still be using the older version due to caching, etc.
